### PR TITLE
[15.0][MIG] sale_coupon_domain_product_discount: Migration to version 15.0

### PR DESCRIPTION
**Pull request by stefan-tecnativa**
cc@Tecnativa TT38308

@chienandalu @pedrobaeza please review!

### CODE REVIEW

**Comment by stefan-tecnativa** [comment]
This module can be splitted, i will commit it when it's done

File: sale_coupon_domain_product_discount/models/coupon_reward.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def _compute_discount_specific_product_ids(self):
         Otherwise it just masks the stored field"""
         domain_product = self.env.context.get("promo_domain_product")
         if domain_product:
-            self.discount_specific_product_ids = self.env["product.product"]._search(
+            self.discount_specific_product_ids = self.env["product.product"].search(
```
Comment (chienandalu) [inline sale_coupon_domain_product_discount/models/coupon_reward.py]: Why?

Comment (stefan-tecnativa) [reply sale_coupon_domain_product_discount/models/coupon_reward.py]: Using "_search" method directly was giving me an error about wrong value of a field.  I tested in this way and worked, but I didn't investigate more the error.

Comment (chienandalu) [reply sale_coupon_domain_product_discount/models/coupon_reward.py]: Well, performance is crucial here so you should find out what that error might be due to.

**Comment by chienandalu** [comment]
You can split it now

File: coupon_domain_product_discount/tests/test_domain_product_discount.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                            "applied_on": "3_global",
+                            "compute_price": "formula",
+                            "base": "list_price",
+                        },
+                    )
+                ],
+            }
+        )
+        # Let's prepare a sale order to play with
+        sale_form = Form(cls.env["sale.order"])
```
Comment (chienandalu) [inline coupon_domain_product_discount/tests/test_domain_product_discount.py]: This declaration should go in the sales module

**Review by chienandalu — CHANGES_REQUESTED** [review]

File: coupon_domain_product_discount/tests/test_domain_product_discount.py (hunk 2 above)
Comment (stefan-tecnativa) [reply coupon_domain_product_discount/tests/test_domain_product_discount.py]: Done

**Review by chienandalu — APPROVED** [review]
Thanks :)

**Review by pedrobaeza — COMMENTED** [review]
The icon.png file is incorrect.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**